### DISCUSSION TITLE
Static homepage and blog page

### DISCUSSION THREAD

**Discussion by gmmedia** [GetPublii/Publii#1578] [Help]
Hello!

I have changed the Front page to a "Selected post" in _Theme > Layout > Front page_.
But how can I set the blog post listing now? Like "/blog/"?

With the actual setup, the readers can't find my blog posts.

thanks
Jochen

**Comment by bobmitro** [comment]
It will be much easier to accomplish with the upcoming release of Publii 0.46, which will have native support for pages. Additionally, you will have the ability to create a prefix for all your posts, such as /blog/....
<img width="956" alt="screenshot 2024-07-23 at 12 49 25" src="https://github.com/user-attachments/assets/3d8b7e54-0115-4470-9c2a-296f83d059de">

**Reply by gmmedia** [reply]
@bobmitro Very cool. Can I test the 0.46 already?

**Reply by bobmitro** [reply]
The .46 version is not bundled yet as an executable (.exe) or disk image (.dmg), but it is available in the .46 branch on GitHub.

**Comment by Backspaze** [comment]
@bobmitro 

**EDIT**: I believe I figured it out. In step four below, in the field **Internal link**, I should've typed **blog/index.html**. I'd still appreciate a built in option called **Posts link** as mentioned in the last paragraph below.

**Old ramble**

Now that 0.46 is released, I tried setting this up, but can't figure out how. After configuring everything, the only thing that happens when I go to example.com/blog/ is that the web browser shows the file/folder structure like a file explorer. What am I missing?

1. I've created a page that will act as my home page.
2. Under **Site settings** > **SEO** > **Homepage**> **Set page as homepage** is enabled, and I've chosen the page from the drop down menu.
3. Under **Site settings** > **URLs** > **Use pretty URLs** is enabled, and I've set **Posts prefix** to **blog**.
4. Under **Menus** I've created a menu item called **Blog** with the type **Internal link** and in the field for **Internal link** I've just typed **blog**.

I'm using the theme [Mono](https://marketplace.getpublii.com/themes/mono/) which has its own setting for this under **Theme** > **Layout** > **Homepage**. There's a note there saying that the feature will be removed in the next theme update, and I've let that setting remain on the default **Latest posts** option.

I'm probably missing something obvious, but it's not working for me. Right now it seems that the only way I can solve this is by creating a page called **Blog** and figure out what code to put there for it to list all blog posts, if that is even possible.

I figure that in a future update, maybe you could add another link type for menu items. There are options for post, page, tag and tags and so on, so why not have an option for post**s** as well? That way we could cut down the four steps above to one.

![bild](https://github.com/user-attachments/assets/50936914-f824-4b4a-9b3d-703488cb6229)

**Reply by bobmitro** [reply]
> I'm probably missing something obvious, but it's not working for me. Right now it seems that the only way I can solve this is by creating a page called Blog and figure out what code to put there for it to list all blog posts, if that is even possible.

By default, the Mono theme displays the posts listing on the homepage. The theme option called "Homepage Displays" allows you to change this behavior, and instead of showing the post listing, you can display the content of a selected post. However, with the Publii v0.46 release, we can now natively set page content on the homepage, making this theme option seem unnecessary.

So, all you need to do to display different content (other than the default post listing) on the homepage is to create a page and mark it in the "Set page as homepage" option in the Site Settings. 

If you still want the post listing to be visible, which will no longer be the case since we replaced it with the page content, you need to assign a prefix (e.g., "blog") in the "Post prefix (pretty URL)" section under Site Settings. 

That's it. Now, the homepage will display the Page content, while the post listing will be available under the URL path you set in the Post prefix earlier.

**Reply by Backspaze** [reply]
Thanks for the quick reply. I'm aware of all the things you mention, but I see that I failed to mention that the issue I ran into only seems to happen when you're previewing the site locally. When the site is synced to a server, the links works as expected without having to add `/index.html` to the link.

I've set up a [lab page](https://backspaze.gitlab.io/lab/) to be able to test these kind of things on a live web page from now on, but it would be great if there was a way to get all links to work even in a local preview.

**Comment by Backspaze** [comment]
Thanks for adding the **Posts index link** in the [Publii Hotfix 0.46.1](https://getpublii.com/blog/release-046.html#hotfixes).
